File: product-webapp/webapp/src/app/model/ActiveSurvey.ts
import { Question } from "./Question";
import { UiSetting } from "./UiSetting";

export class ActiveSurvey
{
    "surveyTitle" : string;
	"description": string;
	"createdBy" : string;
	"status" : string; // ACTIVE/DORMANT
	"activeTime" : number;
	"question" : Question[]; 
    "uiSettings": UiSetting;
    "activatedOn":number;
    "surveyLink":string;
	"targetAudience": string;
	"activeSurveyId": string;
}

File: surveyengine-service/src/main/java/com/stackroute/surveyengineservice/ActiveSurvey/model/ActiveSurvey.java
package com.stackroute.surveyengineservice.ActiveSurvey.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;


import java.util.List;
import java.util.UUID;

@Document(indexName = "activesurvey" , shards= 2)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ActiveSurvey {
    @Id
    public UUID activeSurveyId;
    public String surveyTitle;
    public Status status;
    public  long activeTime;
    public List<Question> question;
    public  String createdBy;
    public String description ;
    public String targetAudience;
    public String surveyLink;
    public long activatedOn;
    public Object uiSettings;

}
